md/variables.md

```diff
@@ -72,4 +72,9 @@ To ease the definition of the form data to send back to the process, when you cr
 * _formInput_. It is a JSON object. Its structure is defined by the contract inputs and the attributes are initialized with default values. It could be used to set initial values for form fields. You can set the values in formInput either by editing the default values with constants (useful for test and debug) or with values from an object in an external source that has the same model (such as with a BDM external API). You can also set the initial values of a form from some other source without using formInput. However, you will then have to edit formOutput manually.
 * _formOutput_. It is a JavaScript expression returning an object. The object structure matches the contract requirements and it is filled with formInput by default. On Submit, values entered or modified by the user and aggregated in the formOutput object (as defined by the _Data sent on click_ property of the Submit button) are submitted to the process or task to fulfill the contract.
 * _taskId_. It is the id of the current BPM task. You can use it as a BPM API parameter.
-* _context_. It is an External API that provides references to all business variables and documents in the process instance. To retrieve a business variable instance, create an External API variable with value `../{{context.myBusinessData_ref.link}}`.
+* _context_. It is an External API that provides references to all business variables and documents in the process instance.
+* Query variables (External API). If data contains aggregation relation, a Select widget is generated with its associated _available values_ data. It uses the BDM API like this `../API/bdm/businessData/com.company.model.Product?q=find&p=0&c=99`. By default it uses the `find` query with a default pagination (only first 100 objects are returned).
+* Data variables (External API). Created in Edit mode only. Data value is retrieved from the _context_ like this `../{{context.myBusinessData_ref.link}}`
+* Lazy reference variables. Created in Edit mode only, when a data contains lazy references. Reference URL is resolved using the _lazyRef_ filter like this `{{invoice|lazyRef:'customer'}}` where `invoice` is a data variables with a lazy reference on the `customer` relation.
+* _submit_errors_list_. It is a JavaScript expression formatting the response payload to html when a submit fails.
```

Comment:
Since this variable is always created, I would put this one above the variables created in some cases.

Reply:
Examples would help